sdks/python/apache_beam/runners/common.py

```diff
@@ -519,11 +705,11 @@ def __init__(self,
         signature.is_stateful_dofn())
     self.user_state_context = user_state_context
     self.is_splittable = signature.is_splittable_dofn()
-    self.watermark_estimator = self.signature.get_watermark_estimator()
-    self.watermark_estimator_param = (
-        self.signature.process_method.watermark_estimator_arg_name
-        if self.watermark_estimator else None)
     self.threadsafe_restriction_tracker = None
+    self.threadsafe_watermark_estimator = None
+    # The lock which guarantee synchronization for both
+    # ThreadsafeRestrictionTracker and ThreadsafeWatermarkEstimator.
+    self._synchronized_lock = threading.RLock()
```

Comment:
can we use threading.Lock()?
(do we really need a re-entrant lock?)

Reply:
Thinking about this again. We don't really need a RLock here. Thanks!